pkg/ccl/changefeedccl/changefeed_stmt.go

```diff
@@ -1362,6 +1401,40 @@ func (b *changefeedResumer) resumeWithRetries(
 	return errors.Wrap(ctx.Err(), `ran out of retries`)
 }
 
+func resolveDest(ctx context.Context, execCfg *sql.ExecutorConfig, sinkURI string) string {
+	u, err := url.Parse(sinkURI)
+	if err != nil || u.Scheme != changefeedbase.SinkSchemeExternalConnection {
+		return sinkURI
+	}
+	resolved := ""
+	if err := execCfg.InternalDB.Txn(ctx, func(ctx context.Context, txn isql.Txn) error {
+		conn, err := externalconn.LoadExternalConnection(ctx, u.Host, txn)
+		if err != nil {
+			return err
+		}
+		resolved = conn.UnredactedConnectionStatement()
+		return nil
+	}); err != nil {
+		log.Warningf(ctx, "failed to resolve external connection %s: %v", u.Host, err)
+	}
+	return resolved
+}
+
+func reloadDetails(
```

Comment:
nit: this function may be renamed `reloadDest` since that's what it returns. Also i don't believe it needs to take the current details as an argument

Reply:
Ah, yeah. I was thinking y'all might be inclined to extend this to reload if _details_ change as well once the poller added here is in-place which could allow relaxing the "must be paused" rule for at least some flavors of ALTER but I guess y'all can rename it when that time comes.